db.tf

```diff
@@ -63,12 +48,18 @@ resource "random_id" "snapshot_id" {
   byte_length = 8
 }
 
+resource "random_string" "db_master_password" {
+  count = "${ var.external_database_host == "" ? 1 : 0 }"
+  length  = 16
+  special = false
+}
+
 resource "aws_rds_cluster" "apiary_cluster" {
   count                               = "${ var.external_database_host == "" ? 1 : 0 }"
   cluster_identifier                  = "${local.instance_alias}-cluster"
   database_name                       = "${var.apiary_database_name}"
-  master_username                     = "${data.vault_generic_secret.apiarydb_master_user.data["username"]}"
-  master_password                     = "${data.vault_generic_secret.apiarydb_master_user.data["password"]}"
+  master_username                     = "${var.db_master_username}"
+  master_password                     = "${random_string.db_master_password.result}"
```

Comment:
this should be optional and we should be able to provide a password from the secret manager.

Reply:
reading individual elements from secretmanager makes it complicated as terraform natively doesn't support json.

If you feel like it is needed you can create PR for this later.

Reply:
We decided to use the master_username as RW user some point. so thought of passing from the secret manager. 
We should store this secret somewhere so that in case we need then we can retrieve the password.

Reply:
can you use RW username instead?

you can retrieve master username from state file, can also reset without impacting anything else if you don't use master username else where.